modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionCategoryServiceImpl.java

```diff
@@ -96,11 +96,14 @@ public CPOptionCategory fetchCPOptionCategory(long cpOptionCategoryId)
 	public CPOptionCategory getCPOptionCategory(long cpOptionCategoryId)
 		throws PortalException {
 
+		CPOptionCategory cpOptionCategory =
+			cpOptionCategoryLocalService.getCPOptionCategory(
+				cpOptionCategoryId);
+
```

Comment:
why this?

Reply:
If you call the API with a not existing optionCategoryId it returns 500 and not 404 (because the check of permissions runs on a not existing id)